ray-curator/ray_curator/stages/text/embedders/base.py

```diff
@@ -0,0 +1,142 @@
+import os
+from dataclasses import dataclass
+from typing import Literal
+
+import cudf
+import cupy as cp
+import pandas as pd
+import torch
+import torch.nn.functional as F  # noqa: N812
+from crossfit.backend.cudf.series import create_list_series_from_1d_or_2d_ar
+from transformers import AutoModel
+
+from ray_curator.backends.base import WorkerMetadata
+from ray_curator.stages.text.models.model import ModelStage
+from ray_curator.stages.text.models.tokenizer import TokenizerStage
+from ray_curator.stages.text.models.utils import ATTENTION_MASK_COLUMN
+
+os.environ["RAPIDS_NO_INITIALIZE"] = "1"
+
+
+from ray_curator.stages.base import CompositeStage, ProcessingStage
+from ray_curator.tasks import DocumentBatch
+
+
+class EmbeddingModelStage(ModelStage):
+    """HuggingFace model stage that produces embeddings with pooling."""
+
+    def __init__(  # noqa: PLR0913
+        self,
+        model_identifier: str,
+        embedding_field: str = "embeddings",
+        pooling: Literal["mean_pooling", "last_token"] = "mean_pooling",
+        hf_token: str | None = None,
+        model_inference_batch_size: int = 256,
+        has_seq_order: bool = True,
+        padding_side: Literal["left", "right"] = "right",
+    ):
+        super().__init__(
+            model_identifier=model_identifier,
+            hf_token=hf_token,
+            model_inference_batch_size=model_inference_batch_size,
+            has_seq_order=has_seq_order,
+            padding_side=padding_side,
+            unpack_inference_batch=True,
+        )
+        self.embedding_field = embedding_field
+        self.pooling = pooling
+
+    def outputs(self) -> tuple[list[str], list[str]]:
+        return ["data"], [self.embedding_field]
+
+    def setup(self, _: WorkerMetadata | None = None) -> None:
```

Comment:
Add a `setup_on_node`?

Reply:
`setup_on_node` is coming from `ModelStage` in `models/model.py`